Make _get_pages function single threaded.

  This fixes qemu segfault caused by race condition

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- python-pip-1.1.orig/pip/index.py
+++ python-pip-1.1/pip/index.py
@@ -221,14 +221,7 @@ class PackageFinder(object):
             pending_queue.put(location)
         done = []
         seen = set()
-        threads = []
-        for i in range(min(10, len(locations))):
-            t = threading.Thread(target=self._get_queued_page, args=(req, pending_queue, done, seen))
-            t.setDaemon(True)
-            threads.append(t)
-            t.start()
-        for t in threads:
-            t.join()
+        self._get_queued_page(req, pending_queue, done, seen)
         return done
 
     _log_lock = threading.Lock()
